### PR TITLE
Fixes #7726 Use WebExtensionRuntime instead of Engine in AddonManager

### LINKED ISSUE #7726: Migrate AddonManager to WebExtensionRuntime
The `AddonManager` is still using engine as the WebExtension management entry point:

https://github.com/mozilla-mobile/android-components/blob/732ac7dcbac61d84c709837d324c3ed46186e6c5/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt#L36-L41

Consider migrating it to use `WebExtensionRuntime` so consumers that not implementing the Concept Engine interface can still use it without needing to provide their own `AddonManager` implementation.

### PR DESCRIPTION
**Pull request by keianhzo**
Fixes #7726 Use WebExtensionRuntime in the AddonManager instead of Engine.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt [hunk 1]
```diff
@@ -9,35 +9,33 @@ import kotlinx.coroutines.CompletableDeferred
...
  * @property addonUpdater The [AddonUpdater] instance to use when checking / triggering
  * updates.
  */
 @Suppress("LargeClass")
 class AddonManager(
-    private val store: BrowserStore,
-    private val engine: Engine,
-    private val addonsProvider: AddonsProvider,
-    private val addonUpdater: AddonUpdater
+        private val store: BrowserStore,
```
Comment (csadilek) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:35]: nit: indentation change, see lint error: Unexpected indentation (expected 4, actual 8) :)

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt [hunk 2]
```diff
@@ -9,35 +9,33 @@ import kotlinx.coroutines.CompletableDeferred
...
 import java.util.Collections.newSetFromMap
 import java.util.concurrent.ConcurrentHashMap
-import java.util.Locale
 
 /**
  * Provides access to installed and recommended [Addon]s and manages their states.
  *
  * @property store The application's [BrowserStore].
- * @property engine The application's browser [Engine].
+ * @property runtime A WebExtensionRuntime instance [WebExtensionRuntime].
```
Comment (csadilek) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:28]: nit: maybe just ```runtime The application's [WebExtensionRuntime] to install and configure extensions.```

File: docs/changelog.md [hunk 3]
```diff
@@ -12,6 +12,9 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Config.kt)
 
+* **feature-addons**
+    * AddonManager now receives a WebExtensionRutime instance instead of an Engine.
```
Comment (csadilek) [inline docs/changelog.md:16]: nit: WebExtensionRu[n]time.

We can also point out that this doesn't not require changes for consumers (as Engine already implements the interface).

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt (hunk 1 above)
Comment (keianhzo) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:35]: Fixed.

Yep, sorry about that, I forgot to add the ticket.

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt (hunk 2 above)
Comment (keianhzo) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:28]: Fixed

File: docs/changelog.md (hunk 3 above)
Comment (keianhzo) [reply docs/changelog.md:16]: Fixed.

**Comment by keianhzo** [comment]
Yep, sorry about that, I forgot to add it and  added it bit later.

**Pull request merged**